src/Azure/Azure.Quantum.Client/JobManagement/CloudJob.cs

```diff
@@ -91,5 +91,16 @@ public async Task CancelAsync(CancellationToken cancellationToken = default)
             CloudJob job = (CloudJob)await this.Workspace.CancelJobAsync(this.Details.Id, cancellationToken);
             this.Details = job.Details;
         }
+
+        private Uri GenerateUri()
+        {
+            if (!(this.Workspace is Workspace cloudWorkspace))
+            {
+                throw new NotSupportedException($"{typeof(CloudJob)}'s Workspace is not of type {typeof(Workspace)} and does not have enough data to generate URI");
+            }
+
+            var uriStr = $"https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/{cloudWorkspace.SubscriptionId}/resourceGroups/{cloudWorkspace.ResourceGroupName}/providers/Microsoft.Quantum/Workspaces/{cloudWorkspace.WorkspaceName}/job_management?microsoft_azure_quantum_jobid={Id}";
```

Comment:
Is there a tests that fails if this uri changes in the future?

Reply:
If the URI of the service changes it does not fail but there various tests would fail if the URI in this implementation changes.

Reply:
Sorry, I was a bit confused, there is no test for that yet. I will add one.

Reply:
This specific test will be added in another repo.